debugger/src/debug_adapter/protocol.rs

```diff
@@ -58,18 +58,32 @@ impl<R: Read, W: Write> DapAdapter<R, W> {
     }
 
     fn send_data(&mut self, raw_data: &[u8]) -> Result<(), std::io::Error> {
-        let response_body = raw_data;
+        let mut response_body = raw_data;
 
         let response_header = format!("Content-Length: {}\r\n\r\n", response_body.len());
 
         self.output.write_all(response_header.as_bytes())?;
+        self.output.flush()?;
 
-        if let Err(err) = self.output.write_all(response_body) {
-            // TODO: Implement catch of errorkind "WouldBlock" and retry
-            tracing::error!("send_data - body: {:?}", err);
-            return Err(err);
+        // NOTE: Sometimes when writing large response, the debugger will fail with an IO error (ErrorKind::WouldBlock == error.kind())
+        let mut bytes_remaining = response_body.len();
+        while bytes_remaining > 0 {
+            match self.output.write(response_body) {
+                Ok(bytes_written) => {
+                    bytes_remaining = bytes_remaining.saturating_sub(bytes_written);
+                    response_body = &response_body[bytes_written..];
+                }
+                Err(error) => {
+                    if error.kind() == std::io::ErrorKind::WouldBlock {
```

Comment:
Maybe we should add a delay here? Not sure if it's a good idea to try immediately again.

Reply:
@Tiwalun ... I tested this with a 5 ms, and with no delay, and it works fine. With even a small delay, there is a noticeable impact on transferring that 48K static the user had, and with no delay, it is as fast as the manual chunking, and no errors. I'm going to elect to leave it without delay for now :)